### PR TITLE
Feat/co 53 show all theme in preview

### PR DESCRIPTION
**Pull request by arjun-at-roam**
## PR Area

- [x] Frontend

## Type of change

_What kind of change does this PR introduce?_

- [x] Feature

## Description

* Assuming that theme is derived from theme URL parameter.
* The idea behind this PR is to make life easier to test all the themes together in one place.
* Run `yarn preview` to test this.

## Security & risks

- [ ] Adds new third party libraries
- [ ] Adds new `.env` variables
- [ ] Integrates with Third-party apis (dependency must be resilient)
- [ ] Has specific [OWASP security concerns](https://owasp.org/www-project-top-ten/) - list them: (also see [Fixing OWASP Top 10](https://nodegoat.herokuapp.com/tutorial))

## Pre-publish checklist

- [x] I have assigned myself to the ticket
- [x] I have performed a self-review of my code
- [x] I have performed relevant browser testing
- [ ] I have performed keyboard testing of new or modified interactive components

#### Evidence screenshots
image

## Reviewer checklist

- [ ] I have run the code (if applicable)
- [ ] I have reviewed and run any tests (if applicable)

### CODE REVIEW

File: src/utils/createDynamicAppAtScriptTag.ts [hunk 1]
```diff
@@ -23,21 +24,30 @@ export const createDynamicAppAtScriptTag = async (
 ): Promise<void> => {
   const { elementID, lang } = config
 
+  const scriptEl = getCurrentScript()
+  const props = getAllScriptURLParameters(scriptEl)
+  const merchantId = props.merchantId
+  const theme = merchantId
```
Comment (ghost) [inline src/utils/createDynamicAppAtScriptTag.ts]: I think `merchantId` is supposed to be a uuid, not a representation of the theme itself. I don't think we're clearly separating concerns if we allow `merchantId` to both be a unique identifier for the merchant as well as a way to set the default theme. Please make `theme` a separate parameter returned from `getAllScriptURLParameters`

File: src/utils/createDynamicAppAtScriptTag.ts [hunk 2]
```diff
@@ -23,21 +24,30 @@ export const createDynamicAppAtScriptTag = async (
...
+
   if (elementID.slice(0, 1) !== '#') {
     throw new Error(
       'elementID must be a valid id selector string. E.g: #my-component'
     )
   }
 
-  const mountTargetSelector = `${elementID.substring(1)}-${lang}`
+  if (theme && !Object.values(ThemeEnum).includes(theme as ThemeEnum))
+    console.warn('theme doesnt exist for this merchantId')
```
Comment (ghost) [inline src/utils/createDynamicAppAtScriptTag.ts]: Please turn this into a block statement as it's easier to read when the condition is so long. Also, you should change this to `throw new Error` - look at the other errors we throw for inspiration on what would provide other developers with a clear error message

**Review by ghost — CHANGES_REQUESTED** [review]

File: public/demo.html [hunk 3]
```diff
@@ -1,19 +1,34 @@
...
   <body>
     <!-- This demo file is used to test both the production bundles after running yarn build -->
     <h1>Humm widgets cards online</h1>
 
-    <h2>AU widget</h2>
+    <h2>AU default widget</h2>
     <script src="./humm-widgets-cards-au.umd.js?productPrice=400"></script>
 
-    <h2>NZ widget</h2>
+    <h2>AU blank widget (remove other widgets to see this in action)</h2>
```
Comment (arjun-at-roam) [inline public/demo.html:16]: blank widgets wont work unless other themed widgets are removed from the page. I've hence added a comment `(remove other widgets to see this in action)`

File: src/utils/createDynamicAppAtScriptTag.ts (hunk 2 above)
Comment (arjun-at-roam) [reply src/utils/createDynamicAppAtScriptTag.ts]: removed this. Back to silently defaulting.

**Review by antoninroam — APPROVED** [review]

**Review by jpyepez-roam — APPROVED** [review]
LGTM!

**Pull request merged**